cmd/argo/commands/copy_artifacts.go

```diff
@@ -0,0 +1,184 @@
+				err = os.MkdirAll(nodeIdFolder, os.ModePerm)
+				if err != nil {
+					log.Fatal(err)
+				}
+				for _, artifact := range artifactList {
+					getAndStoreArtifactData(namespace, copyArtifactArgs.workflowName, nodeId, artifact.Name, nodeIdFolder, c, argoServerArgs)
+				}
+			}
+
+		},
+	}
+	command.Flags().StringVarP(&copyArtifactArgs.namespace, "namespace", "n", "", "namespace of workflow")
+	command.Flags().StringVar(&copyArtifactArgs.workflowName, "workflow-name", "", "name of workflow")
+	command.Flags().StringVar(&copyArtifactArgs.nodeId, "node-id", "", "id of node in workflow")
+	command.Flags().StringVar(&copyArtifactArgs.templateName, "template-name", "", "name of template in workflow")
+	command.Flags().StringVar(&copyArtifactArgs.artifactName, "artifact-name", "", "name of output artifact in workflow")
+	command.Flags().StringVar(&copyArtifactArgs.artifactType, "artifact-type", "output", "type of artifact to copy (only supports output as of now)")
+
+	command.PersistentFlags().StringVarP(&argoServerArgs.URL, "argo-server", "s", os.Getenv("ARGO_SERVER"), "API server `host:port`. e.g. localhost:2746. Defaults to the ARGO_SERVER environment variable.")
+	command.PersistentFlags().StringVar(&argoServerArgs.Path, "argo-base-href", os.Getenv("ARGO_BASE_HREF"), "An path to use with HTTP client (e.g. due to BASE_HREF). Defaults to the ARGO_BASE_HREF environment variable.")
+	command.PersistentFlags().BoolVarP(&argoServerArgs.InsecureSkipVerify, "insecure-skip-verify", "k", os.Getenv("ARGO_INSECURE_SKIP_VERIFY") == "true", "If true, the Argo Server's certificate will not be checked for validity. This will make your HTTPS connections insecure. Defaults to the ARGO_INSECURE_SKIP_VERIFY environment variable.")
+
+	return command
+}
+
+func makeClient(argoServerArgs apiclient.ArgoServerOpts) *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: argoServerArgs.InsecureSkipVerify,
+			},
+		},
+	}
+}
+
+func getAndStoreArtifactData(namespace string, workflowName string, nodeId string, artifactName string, nodeIdFolder string, c *http.Client, argoServerArgs apiclient.ArgoServerOpts) {
+	authString := ""
+	if len(argoServerArgs.URL) == 0 {
+		argoServerArgs.URL = "localhost:2746"
+	} else {
+		authString = client.GetAuthString()
+	}
+	request, err := http.NewRequest("GET", argoServerArgs.GetURL()+fmt.Sprintf("/artifacts/%s/%s/%s/%s", namespace, workflowName, nodeId, artifactName), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if len(authString) != 0 {
+		request.Header.Set("Authorization", client.GetAuthString())
+	}
+	resp, err := c.Do(request)
+	if err != nil {
+		log.Fatalf("Request failed with %s", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		log.Fatalf("Request failed %s", resp.Status)
+	}
+	contentDispositionInfo := resp.Header.Get("Content-Disposition")
+	re := regexp.MustCompile("filename=\"(.*?)\"")
```

Comment:
interesting, I had not considered this, but smart! could do you `artifact.GetKey()` to achive the same?

Reply:
With `artifact.GetKey()` there is a path included (not just filename) that doesn't match the format you had mentioned in the issue. I can use this instead if it is preferred though.

Reply:
filename is just `path.Base(key)`

Reply:
but you're correct, that isn't what I said -- hmmm -- maybe we need a way to specify the file name using templating? Give the user options like kubcetl

```
-f `%nodeId/outputs/%artifactName
```

```
-f "my-file.js"
```

```
-f "%filename"

Reply:
Updated to use artifact.GetKey()